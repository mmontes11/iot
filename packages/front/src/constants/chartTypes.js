export const BARCHART = "BARCHART";
export const LINECHART = "LINECHART";
export const REALTIME = "REALTIME";
