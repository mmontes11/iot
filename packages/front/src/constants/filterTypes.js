export const THING_FILTER_TYPE = "thing";
export const DATE_FILTER_TYPE = "date";
export const FILTER_TYPES = [THING_FILTER_TYPE, DATE_FILTER_TYPE];
