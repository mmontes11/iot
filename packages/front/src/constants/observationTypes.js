export const EVENT_TYPE = "event";
export const MEASUREMENT_TYPE = "measurement";
export const OBSERVATION_TYPES = [EVENT_TYPE, MEASUREMENT_TYPE];
