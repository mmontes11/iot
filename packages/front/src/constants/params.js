export const TYPE = "TYPE";
export const OBSERVATION = "OBSERVATION";
export const GROUPBY = "GROUPBY";
export const THING = "THING";
