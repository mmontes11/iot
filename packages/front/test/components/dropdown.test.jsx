import React from "react";
import { mount } from "enzyme";
import Dropdown from "components/dropdown";
import IntlProvider from "containers/intlProvider";
import { defaultStore } from "../constants";

describe("components/dropdown", () => {
  it("renders in initial state", () => {
    const wrapper = mount(
      <IntlProvider store={defaultStore}>
        <Dropdown
          label="Select item"
          items={["foo", "bar"]}
          isActive={false}
          isLoading={false}
          isDisabled={false}
          onButtonClick={() => undefined}
          onItemClick={() => undefined}
        />
      </IntlProvider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it("renders in active state", () => {
    const wrapper = mount(
      <IntlProvider store={defaultStore}>
        <Dropdown
          label="Select item"
          items={["foo", "bar"]}
          isActive
          isLoading={false}
          isDisabled={false}
          onButtonClick={() => undefined}
          onItemClick={() => undefined}
        />
      </IntlProvider>,
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(".dropdown").hasClass("is-active")).toBeTruthy();
  });
  it("renders in loading state", () => {
    const wrapper = mount(
      <IntlProvider store={defaultStore}>
        <Dropdown
          label="Select item"
          items={["foo", "bar"]}
          isActive={false}
          isLoading
          isDisabled={false}
          onButtonClick={() => undefined}
          onItemClick={() => undefined}
        />
      </IntlProvider>,
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(".button").hasClass("is-loading")).toBeTruthy();
  });
  it("renders in disabled state", () => {
    const wrapper = mount(
      <IntlProvider store={defaultStore}>
        <Dropdown
          label="Select item"
          items={["foo", "bar"]}
          isActive={false}
          isLoading={false}
          isDisabled
          onButtonClick={() => undefined}
          onItemClick={() => undefined}
        />
      </IntlProvider>,
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(".button").prop("disabled")).toBeTruthy();
  });
  it("simulates a click in button", () => {
    const onButtonClick = jest.fn();
    const wrapper = mount(
      <IntlProvider store={defaultStore}>
        <Dropdown
          label="Select item"
          items={["foo", "bar"]}
          isActive={false}
          isLoading={false}
          isDisabled={false}
          onButtonClick={onButtonClick}
          onItemClick={() => undefined}
        />
      </IntlProvider>,
    );
    wrapper.find(".button").simulate("click");
    expect(onButtonClick).toHaveBeenCalledTimes(1);
  });
  it("simulates a click in an item", () => {
    const onClick = jest.fn();
    const wrapper = mount(
      <IntlProvider store={defaultStore}>
        <Dropdown
          label="Select item"
          items={["foo", "bar"]}
          isActive={false}
          isLoading={false}
          isDisabled={false}
          onButtonClick={onClick}
          onItemClick={onClick}
        />
      </IntlProvider>,
    );
    const dropdownItem = wrapper.find(".dropdown-item").first();
    dropdownItem.simulate("click");
    dropdownItem.simulate("keypress", { key: "a" });
    expect(onClick).toHaveBeenCalledTimes(2);
  });
});
