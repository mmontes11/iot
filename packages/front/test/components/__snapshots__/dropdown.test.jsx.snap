// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/dropdown renders in active state 1`] = `
<Connect(IntlProvider)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <IntlProvider
    dispatch={[Function]}
    locale="en"
    messages={
      Object {
        "Custom": "Custom",
        "Data": "Data",
        "Delete": "Delete",
        "End date": "End date",
        "Error": "Error",
        "Event stats": "Event stats",
        "Events:": "Events:",
        "Filters": "Filters",
        "Google maps": "Google maps",
        "Group by": "Group by",
        "IP address:": "IP address:",
        "IoT": "IoT",
        "Last observation:": "Last observation:",
        "Login": "Login",
        "Logout": "Logout",
        "MQTT topic:": "MQTT topic:",
        "Measurement stats": "Measurement stats",
        "Measurement type": "Measurement type",
        "Measurements:": "Measurements:",
        "Observation": "Observation",
        "Password": "Password",
        "Real time": "Real time",
        "Request failed": "Request failed",
        "Reset": "Reset",
        "Start date": "Start date",
        "Stats": "Stats",
        "Thing": "Thing",
        "Thing not found": "Thing not found",
        "Things": "Things",
        "Time": "Time",
        "Time period": "Time period",
        "Type": "Type",
        "Username": "Username",
        "avg": "Average",
        "avgByHour": "Average by hour",
        "date": "Date",
        "day": "Day",
        "door-closed": "Door closed",
        "door-opened": "Door opened",
        "en": "English",
        "es": "Spanish",
        "event": "Event",
        "hour": "Hour",
        "humidity": "Humidity",
        "humidity-indoor": "Humidity indoor",
        "humidity-outdoor": "Humidity outdoor",
        "luminosity": "Luminosity",
        "max": "Maximum",
        "maxByHour": "Maximum by hour",
        "measurement": "Mesasurement",
        "min": "Minimum",
        "minByHour": "Minimum by hour",
        "minute": "Minute",
        "month": "Month",
        "second": "Second",
        "stdDev": "Standard deviation",
        "stdDevByHour": "Standard deviation by hour",
        "temperature": "Temperature",
        "temperature-indoor": "Temperature indoor",
        "temperature-outdoor": "Temperature outdoor",
        "thing": "Thing",
        "total": "Total",
        "week": "Week",
        "year": "Year",
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  >
    <IntlProvider
      key="en"
      locale="en"
      messages={
        Object {
          "Custom": "Custom",
          "Data": "Data",
          "Delete": "Delete",
          "End date": "End date",
          "Error": "Error",
          "Event stats": "Event stats",
          "Events:": "Events:",
          "Filters": "Filters",
          "Google maps": "Google maps",
          "Group by": "Group by",
          "IP address:": "IP address:",
          "IoT": "IoT",
          "Last observation:": "Last observation:",
          "Login": "Login",
          "Logout": "Logout",
          "MQTT topic:": "MQTT topic:",
          "Measurement stats": "Measurement stats",
          "Measurement type": "Measurement type",
          "Measurements:": "Measurements:",
          "Observation": "Observation",
          "Password": "Password",
          "Real time": "Real time",
          "Request failed": "Request failed",
          "Reset": "Reset",
          "Start date": "Start date",
          "Stats": "Stats",
          "Thing": "Thing",
          "Thing not found": "Thing not found",
          "Things": "Things",
          "Time": "Time",
          "Time period": "Time period",
          "Type": "Type",
          "Username": "Username",
          "avg": "Average",
          "avgByHour": "Average by hour",
          "date": "Date",
          "day": "Day",
          "door-closed": "Door closed",
          "door-opened": "Door opened",
          "en": "English",
          "es": "Spanish",
          "event": "Event",
          "hour": "Hour",
          "humidity": "Humidity",
          "humidity-indoor": "Humidity indoor",
          "humidity-outdoor": "Humidity outdoor",
          "luminosity": "Luminosity",
          "max": "Maximum",
          "maxByHour": "Maximum by hour",
          "measurement": "Mesasurement",
          "min": "Minimum",
          "minByHour": "Minimum by hour",
          "minute": "Minute",
          "month": "Month",
          "second": "Second",
          "stdDev": "Standard deviation",
          "stdDevByHour": "Standard deviation by hour",
          "temperature": "Temperature",
          "temperature-indoor": "Temperature indoor",
          "temperature-outdoor": "Temperature outdoor",
          "thing": "Thing",
          "total": "Total",
          "week": "Week",
          "year": "Year",
        }
      }
    >
      <Dropdown
        buttonStyle=""
        iconStyle="fa-angle-down"
        isActive={true}
        isDisabled={false}
        isLoading={false}
        items={
          Array [
            "foo",
            "bar",
          ]
        }
        label="Select item"
        onButtonClick={[Function]}
        onItemClick={[Function]}
      >
        <div
          className="dropdown is-active"
        >
          <div
            className="dropdown-trigger"
          >
            <button
              aria-controls="dropdown-menu"
              aria-haspopup="true"
              className="button"
              disabled={false}
              onClick={[Function]}
            >
              <span>
                <FormattedMessage
                  defaultMessage="Select item"
                  id="Select item"
                  values={Object {}}
                >
                  <span>
                    Select item
                  </span>
                </FormattedMessage>
              </span>
              <span
                className="icon is-small"
              >
                <i
                  aria-hidden="true"
                  className="fas fa-angle-down"
                />
              </span>
            </button>
          </div>
          <div
            className="dropdown-menu"
            id="dropdown-menu"
            role="menu"
          >
            <div
              className="dropdown-content"
            >
              <div
                className="dropdown-item"
                key="foo"
                onClick={[Function]}
                onKeyPress={[Function]}
                role="button"
                tabIndex={0}
              >
                <FormattedMessage
                  defaultMessage="foo"
                  id="foo"
                  values={Object {}}
                >
                  <span>
                    foo
                  </span>
                </FormattedMessage>
              </div>
              <div
                className="dropdown-item"
                key="bar"
                onClick={[Function]}
                onKeyPress={[Function]}
                role="button"
                tabIndex={0}
              >
                <FormattedMessage
                  defaultMessage="bar"
                  id="bar"
                  values={Object {}}
                >
                  <span>
                    bar
                  </span>
                </FormattedMessage>
              </div>
            </div>
          </div>
        </div>
      </Dropdown>
    </IntlProvider>
  </IntlProvider>
</Connect(IntlProvider)>
`;

exports[`components/dropdown renders in disabled state 1`] = `
<Connect(IntlProvider)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <IntlProvider
    dispatch={[Function]}
    locale="en"
    messages={
      Object {
        "Custom": "Custom",
        "Data": "Data",
        "Delete": "Delete",
        "End date": "End date",
        "Error": "Error",
        "Event stats": "Event stats",
        "Events:": "Events:",
        "Filters": "Filters",
        "Google maps": "Google maps",
        "Group by": "Group by",
        "IP address:": "IP address:",
        "IoT": "IoT",
        "Last observation:": "Last observation:",
        "Login": "Login",
        "Logout": "Logout",
        "MQTT topic:": "MQTT topic:",
        "Measurement stats": "Measurement stats",
        "Measurement type": "Measurement type",
        "Measurements:": "Measurements:",
        "Observation": "Observation",
        "Password": "Password",
        "Real time": "Real time",
        "Request failed": "Request failed",
        "Reset": "Reset",
        "Start date": "Start date",
        "Stats": "Stats",
        "Thing": "Thing",
        "Thing not found": "Thing not found",
        "Things": "Things",
        "Time": "Time",
        "Time period": "Time period",
        "Type": "Type",
        "Username": "Username",
        "avg": "Average",
        "avgByHour": "Average by hour",
        "date": "Date",
        "day": "Day",
        "door-closed": "Door closed",
        "door-opened": "Door opened",
        "en": "English",
        "es": "Spanish",
        "event": "Event",
        "hour": "Hour",
        "humidity": "Humidity",
        "humidity-indoor": "Humidity indoor",
        "humidity-outdoor": "Humidity outdoor",
        "luminosity": "Luminosity",
        "max": "Maximum",
        "maxByHour": "Maximum by hour",
        "measurement": "Mesasurement",
        "min": "Minimum",
        "minByHour": "Minimum by hour",
        "minute": "Minute",
        "month": "Month",
        "second": "Second",
        "stdDev": "Standard deviation",
        "stdDevByHour": "Standard deviation by hour",
        "temperature": "Temperature",
        "temperature-indoor": "Temperature indoor",
        "temperature-outdoor": "Temperature outdoor",
        "thing": "Thing",
        "total": "Total",
        "week": "Week",
        "year": "Year",
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  >
    <IntlProvider
      key="en"
      locale="en"
      messages={
        Object {
          "Custom": "Custom",
          "Data": "Data",
          "Delete": "Delete",
          "End date": "End date",
          "Error": "Error",
          "Event stats": "Event stats",
          "Events:": "Events:",
          "Filters": "Filters",
          "Google maps": "Google maps",
          "Group by": "Group by",
          "IP address:": "IP address:",
          "IoT": "IoT",
          "Last observation:": "Last observation:",
          "Login": "Login",
          "Logout": "Logout",
          "MQTT topic:": "MQTT topic:",
          "Measurement stats": "Measurement stats",
          "Measurement type": "Measurement type",
          "Measurements:": "Measurements:",
          "Observation": "Observation",
          "Password": "Password",
          "Real time": "Real time",
          "Request failed": "Request failed",
          "Reset": "Reset",
          "Start date": "Start date",
          "Stats": "Stats",
          "Thing": "Thing",
          "Thing not found": "Thing not found",
          "Things": "Things",
          "Time": "Time",
          "Time period": "Time period",
          "Type": "Type",
          "Username": "Username",
          "avg": "Average",
          "avgByHour": "Average by hour",
          "date": "Date",
          "day": "Day",
          "door-closed": "Door closed",
          "door-opened": "Door opened",
          "en": "English",
          "es": "Spanish",
          "event": "Event",
          "hour": "Hour",
          "humidity": "Humidity",
          "humidity-indoor": "Humidity indoor",
          "humidity-outdoor": "Humidity outdoor",
          "luminosity": "Luminosity",
          "max": "Maximum",
          "maxByHour": "Maximum by hour",
          "measurement": "Mesasurement",
          "min": "Minimum",
          "minByHour": "Minimum by hour",
          "minute": "Minute",
          "month": "Month",
          "second": "Second",
          "stdDev": "Standard deviation",
          "stdDevByHour": "Standard deviation by hour",
          "temperature": "Temperature",
          "temperature-indoor": "Temperature indoor",
          "temperature-outdoor": "Temperature outdoor",
          "thing": "Thing",
          "total": "Total",
          "week": "Week",
          "year": "Year",
        }
      }
    >
      <Dropdown
        buttonStyle=""
        iconStyle="fa-angle-down"
        isActive={false}
        isDisabled={true}
        isLoading={false}
        items={
          Array [
            "foo",
            "bar",
          ]
        }
        label="Select item"
        onButtonClick={[Function]}
        onItemClick={[Function]}
      >
        <div
          className="dropdown"
        >
          <div
            className="dropdown-trigger"
          >
            <button
              aria-controls="dropdown-menu"
              aria-haspopup="true"
              className="button"
              disabled={true}
              onClick={[Function]}
            >
              <span>
                <FormattedMessage
                  defaultMessage="Select item"
                  id="Select item"
                  values={Object {}}
                >
                  <span>
                    Select item
                  </span>
                </FormattedMessage>
              </span>
              <span
                className="icon is-small"
              >
                <i
                  aria-hidden="true"
                  className="fas fa-angle-down"
                />
              </span>
            </button>
          </div>
          <div
            className="dropdown-menu"
            id="dropdown-menu"
            role="menu"
          >
            <div
              className="dropdown-content"
            >
              <div
                className="dropdown-item"
                key="foo"
                onClick={[Function]}
                onKeyPress={[Function]}
                role="button"
                tabIndex={0}
              >
                <FormattedMessage
                  defaultMessage="foo"
                  id="foo"
                  values={Object {}}
                >
                  <span>
                    foo
                  </span>
                </FormattedMessage>
              </div>
              <div
                className="dropdown-item"
                key="bar"
                onClick={[Function]}
                onKeyPress={[Function]}
                role="button"
                tabIndex={0}
              >
                <FormattedMessage
                  defaultMessage="bar"
                  id="bar"
                  values={Object {}}
                >
                  <span>
                    bar
                  </span>
                </FormattedMessage>
              </div>
            </div>
          </div>
        </div>
      </Dropdown>
    </IntlProvider>
  </IntlProvider>
</Connect(IntlProvider)>
`;

exports[`components/dropdown renders in initial state 1`] = `
<Connect(IntlProvider)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <IntlProvider
    dispatch={[Function]}
    locale="en"
    messages={
      Object {
        "Custom": "Custom",
        "Data": "Data",
        "Delete": "Delete",
        "End date": "End date",
        "Error": "Error",
        "Event stats": "Event stats",
        "Events:": "Events:",
        "Filters": "Filters",
        "Google maps": "Google maps",
        "Group by": "Group by",
        "IP address:": "IP address:",
        "IoT": "IoT",
        "Last observation:": "Last observation:",
        "Login": "Login",
        "Logout": "Logout",
        "MQTT topic:": "MQTT topic:",
        "Measurement stats": "Measurement stats",
        "Measurement type": "Measurement type",
        "Measurements:": "Measurements:",
        "Observation": "Observation",
        "Password": "Password",
        "Real time": "Real time",
        "Request failed": "Request failed",
        "Reset": "Reset",
        "Start date": "Start date",
        "Stats": "Stats",
        "Thing": "Thing",
        "Thing not found": "Thing not found",
        "Things": "Things",
        "Time": "Time",
        "Time period": "Time period",
        "Type": "Type",
        "Username": "Username",
        "avg": "Average",
        "avgByHour": "Average by hour",
        "date": "Date",
        "day": "Day",
        "door-closed": "Door closed",
        "door-opened": "Door opened",
        "en": "English",
        "es": "Spanish",
        "event": "Event",
        "hour": "Hour",
        "humidity": "Humidity",
        "humidity-indoor": "Humidity indoor",
        "humidity-outdoor": "Humidity outdoor",
        "luminosity": "Luminosity",
        "max": "Maximum",
        "maxByHour": "Maximum by hour",
        "measurement": "Mesasurement",
        "min": "Minimum",
        "minByHour": "Minimum by hour",
        "minute": "Minute",
        "month": "Month",
        "second": "Second",
        "stdDev": "Standard deviation",
        "stdDevByHour": "Standard deviation by hour",
        "temperature": "Temperature",
        "temperature-indoor": "Temperature indoor",
        "temperature-outdoor": "Temperature outdoor",
        "thing": "Thing",
        "total": "Total",
        "week": "Week",
        "year": "Year",
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  >
    <IntlProvider
      key="en"
      locale="en"
      messages={
        Object {
          "Custom": "Custom",
          "Data": "Data",
          "Delete": "Delete",
          "End date": "End date",
          "Error": "Error",
          "Event stats": "Event stats",
          "Events:": "Events:",
          "Filters": "Filters",
          "Google maps": "Google maps",
          "Group by": "Group by",
          "IP address:": "IP address:",
          "IoT": "IoT",
          "Last observation:": "Last observation:",
          "Login": "Login",
          "Logout": "Logout",
          "MQTT topic:": "MQTT topic:",
          "Measurement stats": "Measurement stats",
          "Measurement type": "Measurement type",
          "Measurements:": "Measurements:",
          "Observation": "Observation",
          "Password": "Password",
          "Real time": "Real time",
          "Request failed": "Request failed",
          "Reset": "Reset",
          "Start date": "Start date",
          "Stats": "Stats",
          "Thing": "Thing",
          "Thing not found": "Thing not found",
          "Things": "Things",
          "Time": "Time",
          "Time period": "Time period",
          "Type": "Type",
          "Username": "Username",
          "avg": "Average",
          "avgByHour": "Average by hour",
          "date": "Date",
          "day": "Day",
          "door-closed": "Door closed",
          "door-opened": "Door opened",
          "en": "English",
          "es": "Spanish",
          "event": "Event",
          "hour": "Hour",
          "humidity": "Humidity",
          "humidity-indoor": "Humidity indoor",
          "humidity-outdoor": "Humidity outdoor",
          "luminosity": "Luminosity",
          "max": "Maximum",
          "maxByHour": "Maximum by hour",
          "measurement": "Mesasurement",
          "min": "Minimum",
          "minByHour": "Minimum by hour",
          "minute": "Minute",
          "month": "Month",
          "second": "Second",
          "stdDev": "Standard deviation",
          "stdDevByHour": "Standard deviation by hour",
          "temperature": "Temperature",
          "temperature-indoor": "Temperature indoor",
          "temperature-outdoor": "Temperature outdoor",
          "thing": "Thing",
          "total": "Total",
          "week": "Week",
          "year": "Year",
        }
      }
    >
      <Dropdown
        buttonStyle=""
        iconStyle="fa-angle-down"
        isActive={false}
        isDisabled={false}
        isLoading={false}
        items={
          Array [
            "foo",
            "bar",
          ]
        }
        label="Select item"
        onButtonClick={[Function]}
        onItemClick={[Function]}
      >
        <div
          className="dropdown"
        >
          <div
            className="dropdown-trigger"
          >
            <button
              aria-controls="dropdown-menu"
              aria-haspopup="true"
              className="button"
              disabled={false}
              onClick={[Function]}
            >
              <span>
                <FormattedMessage
                  defaultMessage="Select item"
                  id="Select item"
                  values={Object {}}
                >
                  <span>
                    Select item
                  </span>
                </FormattedMessage>
              </span>
              <span
                className="icon is-small"
              >
                <i
                  aria-hidden="true"
                  className="fas fa-angle-down"
                />
              </span>
            </button>
          </div>
          <div
            className="dropdown-menu"
            id="dropdown-menu"
            role="menu"
          >
            <div
              className="dropdown-content"
            >
              <div
                className="dropdown-item"
                key="foo"
                onClick={[Function]}
                onKeyPress={[Function]}
                role="button"
                tabIndex={0}
              >
                <FormattedMessage
                  defaultMessage="foo"
                  id="foo"
                  values={Object {}}
                >
                  <span>
                    foo
                  </span>
                </FormattedMessage>
              </div>
              <div
                className="dropdown-item"
                key="bar"
                onClick={[Function]}
                onKeyPress={[Function]}
                role="button"
                tabIndex={0}
              >
                <FormattedMessage
                  defaultMessage="bar"
                  id="bar"
                  values={Object {}}
                >
                  <span>
                    bar
                  </span>
                </FormattedMessage>
              </div>
            </div>
          </div>
        </div>
      </Dropdown>
    </IntlProvider>
  </IntlProvider>
</Connect(IntlProvider)>
`;

exports[`components/dropdown renders in loading state 1`] = `
<Connect(IntlProvider)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <IntlProvider
    dispatch={[Function]}
    locale="en"
    messages={
      Object {
        "Custom": "Custom",
        "Data": "Data",
        "Delete": "Delete",
        "End date": "End date",
        "Error": "Error",
        "Event stats": "Event stats",
        "Events:": "Events:",
        "Filters": "Filters",
        "Google maps": "Google maps",
        "Group by": "Group by",
        "IP address:": "IP address:",
        "IoT": "IoT",
        "Last observation:": "Last observation:",
        "Login": "Login",
        "Logout": "Logout",
        "MQTT topic:": "MQTT topic:",
        "Measurement stats": "Measurement stats",
        "Measurement type": "Measurement type",
        "Measurements:": "Measurements:",
        "Observation": "Observation",
        "Password": "Password",
        "Real time": "Real time",
        "Request failed": "Request failed",
        "Reset": "Reset",
        "Start date": "Start date",
        "Stats": "Stats",
        "Thing": "Thing",
        "Thing not found": "Thing not found",
        "Things": "Things",
        "Time": "Time",
        "Time period": "Time period",
        "Type": "Type",
        "Username": "Username",
        "avg": "Average",
        "avgByHour": "Average by hour",
        "date": "Date",
        "day": "Day",
        "door-closed": "Door closed",
        "door-opened": "Door opened",
        "en": "English",
        "es": "Spanish",
        "event": "Event",
        "hour": "Hour",
        "humidity": "Humidity",
        "humidity-indoor": "Humidity indoor",
        "humidity-outdoor": "Humidity outdoor",
        "luminosity": "Luminosity",
        "max": "Maximum",
        "maxByHour": "Maximum by hour",
        "measurement": "Mesasurement",
        "min": "Minimum",
        "minByHour": "Minimum by hour",
        "minute": "Minute",
        "month": "Month",
        "second": "Second",
        "stdDev": "Standard deviation",
        "stdDevByHour": "Standard deviation by hour",
        "temperature": "Temperature",
        "temperature-indoor": "Temperature indoor",
        "temperature-outdoor": "Temperature outdoor",
        "thing": "Thing",
        "total": "Total",
        "week": "Week",
        "year": "Year",
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  >
    <IntlProvider
      key="en"
      locale="en"
      messages={
        Object {
          "Custom": "Custom",
          "Data": "Data",
          "Delete": "Delete",
          "End date": "End date",
          "Error": "Error",
          "Event stats": "Event stats",
          "Events:": "Events:",
          "Filters": "Filters",
          "Google maps": "Google maps",
          "Group by": "Group by",
          "IP address:": "IP address:",
          "IoT": "IoT",
          "Last observation:": "Last observation:",
          "Login": "Login",
          "Logout": "Logout",
          "MQTT topic:": "MQTT topic:",
          "Measurement stats": "Measurement stats",
          "Measurement type": "Measurement type",
          "Measurements:": "Measurements:",
          "Observation": "Observation",
          "Password": "Password",
          "Real time": "Real time",
          "Request failed": "Request failed",
          "Reset": "Reset",
          "Start date": "Start date",
          "Stats": "Stats",
          "Thing": "Thing",
          "Thing not found": "Thing not found",
          "Things": "Things",
          "Time": "Time",
          "Time period": "Time period",
          "Type": "Type",
          "Username": "Username",
          "avg": "Average",
          "avgByHour": "Average by hour",
          "date": "Date",
          "day": "Day",
          "door-closed": "Door closed",
          "door-opened": "Door opened",
          "en": "English",
          "es": "Spanish",
          "event": "Event",
          "hour": "Hour",
          "humidity": "Humidity",
          "humidity-indoor": "Humidity indoor",
          "humidity-outdoor": "Humidity outdoor",
          "luminosity": "Luminosity",
          "max": "Maximum",
          "maxByHour": "Maximum by hour",
          "measurement": "Mesasurement",
          "min": "Minimum",
          "minByHour": "Minimum by hour",
          "minute": "Minute",
          "month": "Month",
          "second": "Second",
          "stdDev": "Standard deviation",
          "stdDevByHour": "Standard deviation by hour",
          "temperature": "Temperature",
          "temperature-indoor": "Temperature indoor",
          "temperature-outdoor": "Temperature outdoor",
          "thing": "Thing",
          "total": "Total",
          "week": "Week",
          "year": "Year",
        }
      }
    >
      <Dropdown
        buttonStyle=""
        iconStyle="fa-angle-down"
        isActive={false}
        isDisabled={false}
        isLoading={true}
        items={
          Array [
            "foo",
            "bar",
          ]
        }
        label="Select item"
        onButtonClick={[Function]}
        onItemClick={[Function]}
      >
        <div
          className="dropdown"
        >
          <div
            className="dropdown-trigger"
          >
            <button
              aria-controls="dropdown-menu"
              aria-haspopup="true"
              className="button is-loading"
              disabled={false}
              onClick={[Function]}
            >
              <span>
                <FormattedMessage
                  defaultMessage="Select item"
                  id="Select item"
                  values={Object {}}
                >
                  <span>
                    Select item
                  </span>
                </FormattedMessage>
              </span>
              <span
                className="icon is-small"
              >
                <i
                  aria-hidden="true"
                  className="fas fa-angle-down"
                />
              </span>
            </button>
          </div>
          <div
            className="dropdown-menu"
            id="dropdown-menu"
            role="menu"
          >
            <div
              className="dropdown-content"
            >
              <div
                className="dropdown-item"
                key="foo"
                onClick={[Function]}
                onKeyPress={[Function]}
                role="button"
                tabIndex={0}
              >
                <FormattedMessage
                  defaultMessage="foo"
                  id="foo"
                  values={Object {}}
                >
                  <span>
                    foo
                  </span>
                </FormattedMessage>
              </div>
              <div
                className="dropdown-item"
                key="bar"
                onClick={[Function]}
                onKeyPress={[Function]}
                role="button"
                tabIndex={0}
              >
                <FormattedMessage
                  defaultMessage="bar"
                  id="bar"
                  values={Object {}}
                >
                  <span>
                    bar
                  </span>
                </FormattedMessage>
              </div>
            </div>
          </div>
        </div>
      </Dropdown>
    </IntlProvider>
  </IntlProvider>
</Connect(IntlProvider)>
`;
